fvtest/compilertest/tests/SimplifierFoldAndTest.cpp

```diff
@@ -19,54 +19,181 @@
 
-void
-SimplifierFoldAndTest::compileTestMethods()
+/* An IlInjector. This will be used to create the IL that will be compiled.
+ * Note that this can be an IlBuilder, or anything else that acts like an
+ * IlInjector.
+ * Note that this is not an IlGenerator since ResolvedMethod requires an
+ * IlInjector.
+ */
+class SimplifierFoldAndIlInjector : public TR::IlInjector
    {
-   int32_t rc = 0;
+   public:
 
-   TR::TypeDictionary types;
+   TR_ALLOC(TR_Memory::IlGenerator)
 
-   TR::IlType* Int32 = types.PrimitiveType(TR::Int32);
-   TR::IlType* Int64 = types.PrimitiveType(TR::Int64);
+   SimplifierFoldAndIlInjector(TR::TypeDictionary *types, TestDriver *test)
+   :
+      TR::IlInjector(types, test)
+      {
+      }
 
-   int32_t numberOfArguments = 1;
+   bool injectIL()
+      {
+      createBlocks(1);
 
-   TR::IlType** argTypes = new TR::IlType*[numberOfArguments]
-   {
-      Int32
+      TR::SymbolReference* i = newTemp(Int64);
+
+      // int64_t i = ((int64_t) parameter) & 0xFFFFFFFF00000000ll;
+      storeToTemp(i,
+            createWithoutSymRef(TR::land, 2,
+                  iu2l
+                       (parameter(0, Int32)),
+                  lconst(0xFFFFFFFF00000000ll)));
+
+      // return i;
+      returnValue(loadTemp(i));
+
+      return true;
+      }
    };
 
-   SimplifierFoldAndIlInjector ilInjector(&types, this);
+/* Information about the ILInjector. This describes its
+ * signature (name and parameters) and declares the IlInjector.
+ */
+class SimplifierFoldAndInfo : public TestCompiler::MethodInfo
+   {
+   public:
+   SimplifierFoldAndInfo(TestDriver *test)
+   :
+      _ilInjector(&_types, test)
+      {
+      // TODO: make it easier to access Int32/Int64/etc.
+      TR::IlType* Int32 = _types.PrimitiveType(TR::Int32);
+      TR::IlType* Int64 = _types.PrimitiveType(TR::Int64);
+      _args[0] = Int32;
+      DefineFunction(__FILE__, LINETOSTR(__LINE__), "simplifierFoldAnd", 1, _args, Int64);
+      DefineILInjector(&_ilInjector);
+      }
 
-   TR::ResolvedMethod compilee(__FILE__, LINETOSTR(__LINE__), "simplifierFoldAnd", numberOfArguments, argTypes, Int64, 0, &ilInjector);
+   typedef int64_t (*MethodType)(int32_t);
 
-   TR::IlGeneratorMethodDetails details(&compilee);
+   private:
+   TR::TypeDictionary _types;
+   TestCompiler::SimplifierFoldAndIlInjector _ilInjector;
+   TR::IlType *_args[1];
+   };
 
-   testCompiledMethod = reinterpret_cast<SimplifierFoldAndTest::TestCompiledMethodType>(compileMethod(details, warm, rc));
-   }
+/* Verify the optimization was correctly applied.
+ * Errors are added by `ASSERT_*` and `EXPECT_*`.
+ * `ASSERT_*` immediately end compilation, so other
+ * verifiers and functional tests do not run.
+ * `EXPECT_*` only adds an error, which allows the user to
+ * see if there are other failures later.
+ *
+ * While anything can be done in `verify`, this test
+ * is implmented by traversing the IL, and checking if
+ * there are AND operations.
+ */
```

Comment:
Is it just me or is this comment not actually describing the body below (`ASSERT/EXPECT` vs `ADD_FAILURE`)